--- a/zapret2/Makefile
+++ b/zapret2/Makefile
@@ -5,17 +5,17 @@
 include $(TOPDIR)/rules.mk
 
 PKG_NAME:=zapret2
-PKG_VERSION:=0.9.20260307
+PKG_VERSION:=1.0.4
 PKG_RELEASE:=1
 
 PKG_MAINTAINER:=bol-van
 PKG_LICENSE:=MIT
 PKG_LICENSE_FILES:=docs/LICENSE.txt
 
 PKG_SOURCE_URL:=https://github.com/bol-van/zapret2.git
 PKG_SOURCE_PROTO:=git
-PKG_SOURCE_VERSION:=d3b3011000f103c5af161cc4e3167e80fd6928a2
-PKG_SOURCE_DATE:=2026-03-07
+PKG_SOURCE_VERSION:=2c21faa80e1acb71ddceb8b49176f266b7d33f05
+PKG_SOURCE_DATE:=2026-07-31
 
 #PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.gz
 #PKG_SOURCE_URL:=https://github.com/bol-van/zapret2/archive/refs/tags/v$(PKG_VERSION).tar.gz?
